cdap-api/src/main/java/co/cask/cdap/api/Transactional.java

```diff
@@ -34,4 +39,15 @@
    * @throws TransactionFailureException if failed to execute the given {@link TxRunnable} in a transaction
    */
   void execute(TxRunnable runnable) throws TransactionFailureException;
+
+  /**
+   * Executes a set of operations via a {@link TxRunnable} that are committed as a single transaction with a given
+   * timeout. The {@link TxRunnable} can gain access to {@link Dataset} through the provided {@link DatasetContext}.
+   *
+   * @param timeoutInSeconds the transaction timeout for the transaction, in seconds
+   * @param runnable the runnable to be executed in the transaction
+   *
+   * @throws TransactionFailureException if failed to execute the given {@link TxRunnable} in a transaction
+   */
+  void execute(int timeoutInSeconds, TxRunnable runnable) throws TransactionFailureException;
```

Comment:
If someone put in a really long time (e.g. MAX_INT) and actually not committing, will that essentially make the tx manager never be able to cleanup any change sets and eventually runs out of memory? I am just thinking about what's the impact if some app misbehaves (not necessarily intentional, but could be bug).